biosppy/signals/ecg.py

```diff
@@ -1381,6 +1381,858 @@ def ASI_segmenter(signal=None, sampling_rate=1000.0, Pth=5.0):
     return utils.ReturnTuple((rpeaks,), ("rpeaks",))
 
 
+def Pan_Tompkins_Plus_Plus_segmenter(self, ecg, fs):
+    """ ECG QRS-Peak Detection Algorithm 
+
+    Follows the approach by Md Niaz and Naimul [MdNiNai22]
+
+    Parameters
+    ----------
+    
+    ecg : array
+        Input raw ECG signal.
+    fs: int, float, optional
+        Sampling frequency (Hz).
+
+    Returns
+    ----------
+    qrs_i_raw: array
+        R-peak location indices.
+
+    References
+    ----------
+    .. [MdNiNai22] Khan, Naimul and Imtiaz, Md Niaz, "Pan-Tompkins++: A Robust Approach to Detect R-peaks in ECG Signals", 
+        arXiv preprint arXiv:2211.03171, 2022
+
+    """
+
+    # check inputs
+    if ecg is None:
+        raise TypeError("Please specify an input signal.")
+        
+    ''' Initialize '''
+    delay = 0
+    skip = 0                    # Becomes one when a T wave is detected
+    m_selected_RR = 0
+    mean_RR = 0
+    ser_back = 0
+
+
+    ''' Noise Cancelation (Filtering) (5-18 Hz) '''
+
+    if fs == 200:
+        ''' Remove the mean of Signal '''
+        #If fs=200 keep frwquency 5-12 Hz otherwise 5-18 Hz
+        ecg = ecg - np.mean(ecg)  
+
+        Wn = 12*2/fs
+        N = 3
+        a, b = signal.butter(N, Wn, btype='lowpass')
```

Comment:
```suggestion
        a, b = ss.butter(N, Wn, btype='lowpass')
```

Is this function from scipy? If so, the name of the library in this file was imported as ss :)

Reply:
yes, it is from scipy. thanks!